cloud-controller-manager/do/firewall_controller.go

```diff
@@ -290,12 +369,54 @@ func isEqual(targetFirewall *godo.Firewall, fr *godo.FirewallRequest) (bool, []s
 }
 
 func (fm *firewallManager) createFirewall(ctx context.Context, fr *godo.FirewallRequest) (*godo.Firewall, error) {
-	currentFirewall, _, err := fm.client.Firewalls.Create(ctx, fr)
+	currentFirewall, err := func() (*godo.Firewall, error) {
```

Comment:
nit: I think the reason you're using a callback approach in `firewallManager.Get` is that more is happening in that function than simply calling out to the firewalls API whereas here in `firewallManager.createFirewall` we are simply wrapping a single call to the firewalls API; removing the callback would lead to a somewhat cleaner implementation here.